Mathlib/Data/Nat/Prime/Defs.lean

```diff
@@ -137,6 +137,17 @@ theorem prime_def_le_sqrt {p : ℕ} : Prime p ↔ 2 ≤ p ∧ ∀ m, 2 ≤ m →
           refine this km (Nat.lt_of_mul_lt_mul_right (a := m) ?_) e
           rwa [one_mul, ← e]⟩
 
+theorem prime_iff_not_exists_mul_eq {p : ℕ} :
```

Comment:
There are several `not_prime` theorems, and I wonder whether this one shouldn't be worded similarly (`not_prime_iff_exists…`). On the other hand, this is the one you will need.

Reply:
I could add a `not_prime` version next to [Nat.not_prime_iff_exists_dvd_lt](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Nat/Prime/Basic.html#Nat.not_prime_iff_exists_dvd_lt) with `2 ≤ p` as a hypothesis ...